doc/hardware/peripherals/mspi.rst

```diff
@@ -0,0 +1,198 @@
+.. _mspi_api:
+
+Multi-bit SPI Bus
+#################
+
+The MSPI (multi-bit SPI) is provided as a generic API to accommodate
+advanced SPI peripherals and devices that typically require command,
+address and data phases, and multiple signal lines during these phases.
+While the API supports advanced features such as :term:`XIP` and scrambling,
+it is also compatible with generic SPI.
+
+.. contents::
+    :local:
+    :depth: 2
+
+.. _mspi-controller-api:
+
+MSPI Controller API
+*******************
+
+Zephyr's MSPI controller API may be used when a multi-bit SPI controller
+is present. E.g. Ambiq MSPI, QSPI, OSPI, Flexspi, etc.
+The API supports single to hex SDR/DDR IO with variable latency and advanced
+features such as :term:`XIP` and scrambling. Applicable devices include but
+not limited to high-speed, high density flash/psram memory devices, displays
+and sensors.
```

Comment:
Do we really want to expand to sensors ?

Reply:
ok, this was not my understanding while it was discussed (in order to distinguish with SPI API), but if this is fine with everyone, so be it.

Reply:
I'd note that without supplying an ISR safe bus implementation that works as RTIO expects, sensor drivers in Zephyr will not work with this bus.

For sensors, any bus implementation will need to accept and start asynchronous bus transactions from an ISR. Otherwise we are back to dealing with threads/work queues per sensor which gets ridiculous quickly.

Reply:
> Sure, for sensors that have cmd address and data phases even though they may not be multi-bit. The API is also compatible with RTIO if the controller is implemented in a certain way.

I don't see any form of submit API here so no I don't think that's true, and I don't think this API will work with the direction sensors are going today without changes.

Reply:
> > Sure, for sensors that have cmd address and data phases even though they may not be multi-bit. The API is also compatible with RTIO if the controller is implemented in a certain way.
> 
> I don't see any form of submit API here so no I don't think that's true, and I don't think this API will work with the direction sensors are going today without changes.

The documentation saying that the API supports the communication interface required by these categories of devices. The sensors here don’t necessarily means the sensor subsystem.

You are correct, an additional API is needed to support RTIO. In addition, the compatibility will also depend on the controller implementation. 

I have no plan to add that in this PR as we don’t have a sensor to test that. But if we do, the RTIO path will be followed.